### DISCUSSION TITLE
vercel's HOBBY plan , generating pages error , graphQL , speed up

### DISCUSSION THREAD

**Discussion by arashesnaashari** [vercel/next.js#20970] [Help] [answered]
Hello, I'm creating an e_commerce/blog website and using ```SSG``` for products pages

at first, I generate a small subset of pages and set ```fallback: true``` in getStaticPath and ```revalidate:1``` in getStaticProps
also, I'm using ```graphql``` as API and ```mongodb``` as a database **with no use of a custom server**

I deployed it on vercel with ```HOBBY``` plan
as the documentation says :
> Serverless Function Execution Timeout is 10s

when I request a page that's not generated yet , it doesn't return ```router.isFallback```  and after 10s this error appears :

>An error occurred with this application.
>NO_RESPONSE_FROM_FUNCTION

 even in ```localhost:3000``` it takes too long to respond 

 I even tried ```SSR``` but its also took whiles

##### if there wasn't  SSG with all route generated, my app was so slow 

I'm sure I'm doing nextjs SSG and SSR implementation right 

I guess my vercel plan could make this issue but SSR and generating page by user's request is slow even in localhost

does it because I'm using graphql or vercel (HOBBY) cause this?

```javascript
export const getStaticPaths = async () => {
  const blogsData= await queryGraphQl(`query {
    blogs(limit:1){
     _id
  }
}`);
  const paths = blogsData.blogs.map((e) => ({
    params: {
      _id: e._id,
    },
  }));
  return { paths: paths, fallback: true };
};

export const getStaticProps = async ({ params: { _id } }) => {
  const data = await queryGraphQl(`
  query {
    blog(_id:"${_id}") {
      ...
      ...
  }
  }
  `);

  return {
    props: { data: data},
    revalidate: 1,
  };
};
```

and the only error that repeats with every generates req in vercel's function part : 
```2021-01-11T09:43:02.049Z f396c4fb-2a13-4602-8bc4-47d079bc355b Task timed out after 10.01 seconds```

thanks for helping ❤

**Comment by michrome** [comment, marked as answer]
On the Vercel Hobby plan, [your functions must complete in less than 10 seconds](https://vercel.com/docs/platform/limits#general-limits). You’ll get a 60 seconds if you move to the Pro plan.

You mentioned things run slowly even on your local machine, so my recommendation would be to dig into your function to understand what is causing it to take longer than 10 seconds to execute. Is it the Mongo query?

**Reply by arashesnaashari** [reply]
first thanks for your quick answer.
if I upgrade my vercel plan it would fix.
yes, it is.
POST (create new) request are fast enough and the problem is in GET req
I'm trying to improve my mongo query as you mention 
 ### Does the Pro plan increase the speed of response either?

**Reply by michrome** [reply]
No, a paid-for Vercel plan won't speed things up for you from what you describe (the bottleneck is Mongo not Vercel).

That said, if both of the following are true then a paid-for Vercel plan will at least allow your functions to run to completion.

- you can optimise your Mongo query below 60 seconds **AND**
- you **can't** optimise your Mongo query below 10 seconds.

**Reply by arashesnaashari** [reply]
as I understood, 

 1)  optimizing query and 

 2) upgrading the plan

 will fix the ***online_static_generation*** and other issue

thanks so much
